platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Mellanox
+#
+# Module contains an implementation of SONiC Platform Base API and
+# provides the Chassis information which are available in the platform
+#
+#############################################################################
+
+import sys
+
+try:
+    from sonic_platform_base.chassis_base import ChassisBase
+    from sonic_platform.psu import Psu
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+
+MLNX_NUM_PSU = 2
+
+class Chassis(ChassisBase):
+    """Platform-specific Chassis class"""
+    def __init__(self):
+        ChassisBase.__init__(self)
+        self._psu_list.append(None)
+        for index in range(1, MLNX_NUM_PSU + 1):
```

Comment:
I debated changing this from being 1-based to being 0-based. The reasoning is that currently PSUs are referenced as 1-based, whereas other components (transceivers, etc.) are 0-based, so I thought of converting everything to 0-based for consistency. However, we can keep PSUs as 1-based if we think it's necessary. If so, we should also modify the comments here and in sonic-platform-common appropriately, because as you can see below, I mention that `get_psu()` takes a 0-based value.